openid-connect-server/src/main/java/org/mitre/oauth2/web/DeviceEndpoint.java

```diff
@@ -160,12 +161,20 @@ public String requestDeviceCode(@RequestParam("client_id") String clientId, @Req
 
 	@PreAuthorize("hasRole('ROLE_USER')")
 	@RequestMapping(value = "/" + USER_URL, method = RequestMethod.GET)
-	public String requestUserCode(ModelMap model) {
+	public String requestUserCode(@RequestParam(value = "user_code", required = false) String userCode, ModelMap model, HttpSession session) {
```

Comment:
Would it make sense to have a flag somewhere enabling this behavior? Perhaps per-client? There's no standard for that flag so maybe just adding it directly like this can work for now.